GeeksCoreLibrary/Components/Configurator/Services/ConfiguratorsService.cs

```diff
@@ -845,6 +1247,21 @@ public async Task<string> ReplaceConfiguratorItemsAsync(string template, VueConf
             {
                 return template;
             }
+            
+            // Replace external configurator ID if one is provided.
+            if (configuration.ExternalConfiguration != null && !String.IsNullOrWhiteSpace(configuration.ExternalConfiguration.Id))
+            {
+                if (!isDataQuery)
+                {
+                    template = template.Replace($"{{externalConfiguration.id}}", configuration.ExternalConfiguration.Id);
+                }
+                else
+                {
+                    var parameterName = DatabaseHelpers.CreateValidParameterName("externalConfiguration.id");
```

Comment:
You shouldn't call `DatabaseHelpers.CreateValidParameterName` with a static value, just make it a valid parameter name manually and use that.